src/Pulsar.Client/Internal/HttpLookupService.fs

```diff
@@ -0,0 +1,217 @@
+                        Max = TimeSpan.FromMinutes(1.0)
+                    }
+                let! result = this.GetPartitionedTopicMetadataInner(topicName, backoff, int config.OperationTimeout.TotalMilliseconds)
+                return result
+            }
+
+        member this.GetBroker(topicName : CompleteTopicName) =
+            this.GetBrokerInner(topicName)
+
+        member this.GetTopicsUnderNamespace (ns: NamespaceName, isPersistent: bool) =
+            backgroundTask {
+                let backoff =
+                    Backoff {
+                        Initial = TimeSpan.FromSeconds(5.0)
+                        MandatoryStop = config.OperationTimeout + config.OperationTimeout
+                        Max = TimeSpan.FromMinutes(1.0)
+                    }
+                let! result = this.GetTopicsUnderNamespaceInner(ns, backoff, int config.OperationTimeout.TotalMilliseconds, isPersistent)
+                return result
+            }
+
+        member this.GetSchema(topicName: CompleteTopicName, ?schemaVersion: SchemaVersion) =
+            backgroundTask {
+                let backoff =
+                    Backoff {
+                        Initial = TimeSpan.FromSeconds(5.0)
+                        MandatoryStop = config.OperationTimeout + config.OperationTimeout
+                        Max = TimeSpan.FromMinutes(1.0)
+                    }
+                let! result = this.GetSchemaInner(topicName, schemaVersion, backoff, int config.OperationTimeout.TotalMilliseconds)
+                return result
+            }
+
+    //  GET /admin/v2/{topic-domain}/{tenant}/{namespace}/{topic}/partitions?checkAllowAutoCreation=true
+    member private this.GetPartitionedTopicMetadataInner (topicName: CompleteTopicName, backoff: Backoff, remainingTimeMs) =
+         async {
+            try
+                let randomServiceUri = config.ServiceAddresses[RandomGenerator.Next(0, config.ServiceAddresses.Length)]
+                let topic: string = %topicName
+                let topicRestPath = topic.Replace("persistent://","persistent/").Replace("non-persistent://","non-persistent/")
+                let! response = randomServiceUri.AbsoluteUri + $"admin/v2/%s{topicRestPath}/partitions?checkAllowAutoCreation=true"
+                                |> httpClient.GetStreamAsync
+                                |> Async.AwaitTask
+                let brokerResponse = JsonSerializer.Deserialize<PartitionedTopicMetadata>(response, jsonOptions)
+                return brokerResponse
+
+            with Flatten ex ->
+                let nextDelay = Math.Min(backoff.Next(), remainingTimeMs)
+                if nextDelay <= 0 then
+                    reraize ex
+                Log.Logger.LogWarning(ex, "GetPartitionedTopicMetadata failed will retry in {0} ms", nextDelay)
+                do! Async.Sleep nextDelay
+                return! this.GetPartitionedTopicMetadataInner(topicName, backoff, remainingTimeMs - nextDelay)
```

Comment:
Here and in other places - do we want to retry any error? Does java retries every error possible?

Reply:
Java client has a retry strategy in `org/apache/pulsar/client/impl/ConnectionHandler.reconnectLater()` to call getBroker() function, which will retry every error.
https://github.com/apache/pulsar/blob/966a396007aba232357170954fb866e50169bdd6/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionHandler.java#L164

I think other lookup functions like `getSchema()`  in java don't have retry strategy, but it is better to use one. When we rolling update broker cluster, or one broker crash for some reason, the lookup service can still pick another broker service address from serviceUrls to reach broker cluster using retry strategy.